src/Illuminate/Collections/LazyCollection.php

```diff
@@ -1385,4 +1401,14 @@ protected function passthru($method, array $params)
             yield from $this->collect()->$method(...$params);
         });
     }
+
+    /**
+     * Get the current time.
+     *
+     * @return int
+     */
+    protected function now()
+    {
```

Comment:
would a setter/wither metod accepting the `datetime interface` be a better fit here? - hence, we would not need to allow for mockying protected methods.

Reply:
I'd rather have the complexity in the tests than in the implementation.